error/s2n_errno.h

```diff
@@ -218,5 +219,28 @@ extern __thread const char *s2n_debug_str;
 #define S2N_ERROR_IF( cond , x ) do { if ( cond ) { S2N_ERROR( x ); }} while (0)
 #define S2N_ERROR_IF_PTR( cond , x ) do { if ( cond ) { S2N_ERROR_PTR( x ); }} while (0)
 
+#define S2N_PRECONDITION( cond ) S2N_ERROR_IF(!(cond), S2N_ERR_PRECONDITION_VIOLATION)
+#define S2N_PRECONDITION_PTR( cond ) S2N_ERROR_IF_PTR(!(cond), S2N_ERR_PRECONDITION_VIOLATION)
```

Comment:
You don't appear to be using this second form at all, so I'm not sure what it does.

Does it make sense to use a different error string for each of these two macros?

Reply:
It returns a `NULL` pointer, for functions which return `null` instead of `-1` to report errors.

Reply:
They should use the same error string, since they're reporting the same error, same as is done with the `S2N_ERROR_IF` macro above.